rclpy/rclpy/executors.py

```diff
@@ -288,6 +288,9 @@ def spin_until_future_complete(self, future: Future, timeout_sec: float = None)
             end = start + timeout_sec
             timeout_left = timeout_sec
 
+            # Make sure the future wakes this executor when it is done
+            future.add_done_callback(lambda x: self.wake())
```

Comment:
isn't this also needed also in the if case?
Maybe, it can be added above line L283

Reply:
oops, moved it above line 283 in 
62c2ae4